packages/concerto-cto/lib/parser.pegjs

```diff
@@ -1064,6 +1065,104 @@ ConceptDeclaration
       return result;
     }
 
+BooleanScalar
+   = BooleanType __ d:BooleanDefault? __ {
+      return {
+        $class: "concerto.metamodel@1.1.0.BooleanScalar",
+        defaultValue: (d === 'true' ? true : false)
+      };
+  }
+
+IntegerScalar
+   = IntegerType __ d:IntegerDefault? __ range:IntegerDomainValidator? __ {
+      const result = {
+        $class: "concerto.metamodel@1.1.0.IntegerScalar",
+      };
+      // TODO: Enforce that the supertype either has a default or a validator value
```

Comment:
I don't understand the TODO here. Are you saying that is a scalar has domain validator then its "super type" should also have a domain or range?

`scalar MyInteger extends Integer range [0,10]` seems fine.

We don't support scalars extending other scalars do we?

Reply:
Apologies, the todo is no longer applicable. Addressed ✅
Currently scalars are not extendable. Reviewed ✅